soc/arm/st_stm32/stm32l0/soc.c

```diff
@@ -52,6 +52,13 @@ static int stm32l0_init(const struct device *arg)
 	LL_APB1_GRP1_EnableClock(LL_APB1_GRP1_PERIPH_PWR);
 	LL_PWR_SetRegulVoltageScaling(LL_PWR_REGU_VOLTAGE_SCALE1);
 
+	/* On STM32L0, there are some hardfault when enabling DBGMCU bit:
+	 * Sleep, Stop or Standby. See #37119
+	 * For unclear reason, enabling DMA clock fixes this issue
+	 * (similarly than it fixes #34324)
+	 */
+	__HAL_RCC_DMA1_CLK_ENABLE();
```

Comment:
Wouldn't it be better to use a LL call here? Something like LL_AHB1_GRP1_EnableClock(LL_AHB1_GRP1_PERIPH_DMA1);

Reply:
would it be safe to keep DMA reset ?

Reply:
Not a major difference, but OK. done

Reply:
> would it be safe to keep DMA reset ?

Keeping reset signal of the DMA sound to me like an intermediate state, and it is not supposed to be kept hold.
So I think it is safer to release this signal. It is the way I tested, and it is working well up to now.
Lets keep it